src/ol_openedx_course_sync/ol_openedx_course_sync/utils.py

```diff
@@ -136,3 +149,135 @@ def update_default_tabs(source_course_key, target_course_key, user):
+        return has_changes
+
+    fields_to_sync = {
+        # `DiscussionsConfiguration` fields excluding context_key and history.
+        DiscussionsConfiguration: [
+            "enabled",
+            "posting_restrictions",
+            "lti_configuration",
+            "enable_in_context",
+            "enable_graded_units",
+            "unit_level_visibility",
+            "plugin_configuration",
+            "provider_type",
+        ],
+        # `CourseDiscussionSettings` fields excluding course_id and discussions_id_map.
+        CourseDiscussionSettings: [
+            "always_divide_inline_discussions",
+            "reported_content_email_notifications",
+            "division_scheme",
+            "_divided_discussions",
+        ],
+    }
+
+    source_discussions_settings = CourseDiscussionSettings.get(source_course_key)
+    target_discussions_settings = CourseDiscussionSettings.get(target_course_key)
+    _ = sync_model_objects(
+        source_discussions_settings,
+        target_discussions_settings,
+        fields_to_sync[CourseDiscussionSettings],
+    )
+
+    source_discussions_config = DiscussionsConfiguration.get(source_course_key)
+    target_discussions_config = DiscussionsConfiguration.get(target_course_key)
+    has_configuration_changes = sync_model_objects(
+        source_discussions_config,
+        target_discussions_config,
+        fields_to_sync[DiscussionsConfiguration],
+    )
+
+    if not has_configuration_changes:
+        return
+
+    # update discussion settings in modulestore
+    source_course = modulestore().get_course(source_course_key)
+    target_course = modulestore().get_course(target_course_key)
+
+    target_course.discussions_settings["enable_in_context"] = (
+        target_discussions_config.enable_in_context
+    )
+    target_course.discussions_settings["enable_graded_units"] = (
+        target_discussions_config.enable_graded_units
+    )
+    target_course.discussions_settings["unit_level_visibility"] = (
+        target_discussions_config.unit_level_visibility
+    )
+    target_course.discussion_blackouts = source_course.discussion_blackouts
+    target_course.discussion_topics = source_course.discussion_topics
+    modulestore().update_item(target_course, user.id)
+
+
+def get_course_sync_service_user():
+    """
+    Retrieve the service user for course sync operations.
+
+    Returns:
+        User: The service user object.
+    """
+    cache_key = get_cache_key(
+        course_sync_service_worker=settings.OL_OPENEDX_COURSE_SYNC_SERVICE_WORKER_USERNAME
+    )
+    cache_value = TieredCache.get_cached_response(cache_key)
+    if not cache_value.is_found:
+        user = User.objects.filter(
+            username=settings.OL_OPENEDX_COURSE_SYNC_SERVICE_WORKER_USERNAME
+        ).first()
+        TieredCache.set_all_tiers(cache_key, user)
+    else:
+        user = cache_value.value
+
+    return user
+
+
+def should_perform_sync(course_key):
```

Comment:
How about we rename it to `get_syncable_course_runs` which returns a list. It could be an empty list by default or a populated one based on the query filter. That way, we can omit the boolean return type from here. Our check in the calling methods can be based on if a list if empty or not.